pkg/endpoint/bpf.go

```diff
@@ -1406,14 +1427,38 @@ func (e *Endpoint) startSyncPolicyMapController() {
 					return controller.NewExitReason("Endpoint disappeared")
 				}
 				defer e.unlock()
-				return e.syncPolicyMapWithDump()
+				defer e.swapPolicyMapApplied()
+
+				err := e.syncPolicyMapWithDump(ctx)
+				if err != nil {
+					return err
+				}
+
+				// Apply policy changes to L7 proxy
+				proxyWaitGroup := completion.NewWaitGroup(ctx)
+
+				// Ignoring the revertFunc; keep all successful changes even if some fail.
```

Comment:
@jrajahalme what do you think about the move of the `updateNetworkPolicy` here from `EndpointManager.UpdatePolicyMaps`? And do you think we should return the error here so that the controller will keep retrying this?

Reply:
Can you be more specific? `EndpointManager.UpdatePolicyMaps`currently does not call `updateNetworkPolicy`, but it calls `ep.ApplyPolicyMapChanges`.

Reply:
Oops sorry, I meant moving it from `Endpoint.ApplyPolicyMapChanges` to here. Effectively this means we'll call `updateNetworkPolicy` not only on the incremental path, but also during the periodic full reconciliation. So to rephrase the question: is it OK to call `updateNetworkPolicy` every 15 minutes even if there's no changes, or do we need some extra around this to avoid unnecessary calls?

Second question: should the 'sync-policy-map' controller retry failed calls to `updateNetworkPolicy`?